### PR TITLE
added snow_age viirs composite & lookup table enhancement

### PR DESCRIPTION
**Pull request by eysteinn**
Converted snow age composite from mpop
Added generic lookup table enhancement
Added specific snow age lookup table to yaml

### CODE REVIEW

File: satpy/enhancements/__init__.py [hunk 1]
```diff
@@ -48,3 +48,23 @@ def cira_stretch(img, **kwargs):
         chn -= np.log10(0.0223)
         chn /= 1.0 - np.log10(0.0223)
         chn /= 0.75
+
+
+def lookup(img, **kwargs):
```
Comment (mraspaud) [inline satpy/enhancements/__init__.py]: Could you add support for 1- and 4-channels images ? And maybe add a small unit test ?

File: satpy/enhancements/__init__.py [hunk 2]
```diff
@@ -48,3 +48,15 @@ def cira_stretch(img, **kwargs):
...
+    """Assigns values to channels based on a table"""
+    luts = np.array(kwargs['luts'], dtype=np.float32) / 255.0
+
+    for idx, ch in enumerate(img.channels):
+        np.ma.clip(ch, 0, 255, ch)
```
Comment (pnuu) [inline satpy/enhancements/__init__.py]: Maybe make the in-place explicit with `out=ch`?

File: satpy/etc/enhancements/generic.yaml [hunk 3]
```diff
@@ -405,3 +405,108 @@ enhancements:
         stretch: crude
         min_stretch: [0]
         max_stretch: [0.075]
+  snow_age_default:
+    standard_name: snow_age
+    operations:
+    - name: snow_age
+      method: !!python/name:satpy.enhancements.lookup
+      kwargs:
+       luts: [[0, 0, 0], [1, 2, 2], [3, 8, 5], [4, 12, 8], [6, 15, 10], [8, 18, 13], [9, 21, 16],
```
Comment (pnuu) [inline satpy/etc/enhancements/generic.yaml]: Is it possible to have the LUT in a separate file, like .npy, .npz or ascii?

Comment (pnuu) [reply satpy/etc/enhancements/generic.yaml]: Disregard the previous comment on the LUT, in this case it's practically static data and is OK to have it in the YAML. The `lookup()` function is ignorant about the source anyway, so can be used for other things, too.

**Pull request merged**